### PR TITLE
call fields delete endpoint from client only for existing fields

### PR DESCRIPTION
**Pull request by Tayef-Shah**
### Notion ticket link

[Delete fields](https://www.notion.so/uwblueprintexecs/Allow-users-to-delete-fields-331d40a269894065b8d60bf3dc1e45d2)

### Implementation description

- add destroy action to API utils (didn't use delete since it's a keyword)
- make a function to call delete field endpoint
- call deleteField from FieldEditor

### Steps to test

1. run backend on https://github.com/uwblueprint/project-read-backend/pull/124
2. go to configure global questions
2. click on pencil next to a dynamic field
3. click on trash can to delete
4. go to Django admin and check the field is marked as deleted

### What should reviewers focus on?

- testing
- completeness (what else would be good to add?)

### Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] I have run the linter
- [x] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by brittanylau — COMMENTED** [review]
code looks good for deleting existing fields in the db! i just merged #115, where we updated it so that `field` may not have an id if a user is currently in the process of creating it. we may need to add a check so that we only call the delete endpoint if the field actually exists in the db (`existingFieldId` is not null)

do you have bandwidth incorporate deleting a field that users are currently creating? if not, let's just merge this with the `existingFieldId` check mentioned above, and we can make a ticket for me to do :)

File: src/api/APIUtils.ts [hunk 1]
```diff
@@ -67,3 +67,19 @@ export const put = async (path: string, data: unknown): Promise<unknown> => {
 
   return res.json();
 };
+
+export const destroy = async (path: string): Promise<unknown> => {
```
[the remark this answers is not in the document]
Comment (Tayef-Shah) [reply src/api/APIUtils.ts:71]: delete is a keyword, so it didn't let me name it delete :/

**Comment by Tayef-Shah** [comment]
> do you have bandwidth incorporate deleting a field that users are currently creating? if not, let's just merge this with the `existingFieldId` check mentioned above, and we can make a ticket for me to do :)

I got some time, so I'll give it a shot. Will let you know if I can't make it work

**Comment by Tayef-Shah** [comment]
@brittanylau I added the 'existingId' check, but I'm not sure how to delete a field that is currently being created. Would you be able to take a stab at it?

**Comment by brittanylau** [comment]
> @brittanylau I added the `existingId` check, but I wasn't able to figure out how to delete a field that is currently being created. Would you be able to take a stab at it? 😬

@Tayef-Shah yeah can do, could you make a quick ticket in notion and assign me to it?

**Review by brittanylau — APPROVED** [review]

**Comment by Tayef-Shah** [comment]
> @Tayef-Shah yeah can do, could you make a quick ticket in notion and assign me to it?...
Here's the [ticket ](https://www.notion.so/uwblueprintexecs/Allow-users-to-delete-fields-currently-being-created-10b8435ac0f5474b9ee8db2de0bb2fe7) 🙏 🙏 🙏

**Pull request merged**